app/experimenter/experiments/tests/factories/nimbus.py

```diff
@@ -170,6 +170,30 @@ def create_with_status(cls, target_status, **kwargs):
 
         return NimbusExperiment.objects.get(id=experiment.id)
 
+    @classmethod
+    def create_with_lifecycle(
+        cls,
+        lifecycle=NimbusExperiment.Lifecycles.CREATED,
+        **kwargs,
+    ):
+        experiment = cls.create(**kwargs)
+
+        for state in lifecycle.value:
+            for name, value in state.value.items():
+                setattr(experiment, name, value)
+            experiment.save()
+
+            if experiment.has_filter(experiment.Filters.SHOULD_ALLOCATE_BUCKETS):
+                experiment.allocate_bucket_range()
+
+            generate_nimbus_changelog(
+                experiment,
+                experiment.owner,
+                f"set lifecycle {lifecycle} state {state}",
+            )
+
+        return NimbusExperiment.objects.get(id=experiment.id)
```

Comment:
This is 100% exactly what I was imaginging so yes you exactly nailed it.  The only thing I'd question is whether to have the CREATED as the default.  On the one hand, yeah totally, seems like a safe default.  On the other hand, at the call sites in the tests, reading just `create_with_lifecycle()` doesn't make it exactly clear to me what state it's gonna wind up in, reading it I'd like to always see the target state.  But CREATED is a sensible default so, could probably go either way on it.

Reply:
Yeah, I might get rid of the default just to be explicit everywhere